js/components/ViewPagerIndicator.js

```diff
@@ -0,0 +1,84 @@
+import React, { PropTypes } from 'react';
+import {
+  View,
+  Animated,
+  Dimensions,
+  StyleSheet,
+} from 'react-native';
+
+const { width } = Dimensions.get('window');
+const DOT_SIZE = 12;
+const DOT_SPACING = DOT_SIZE / 2;
+const DOT_FULL_WIDTH = DOT_SIZE + DOT_SPACING * 2;
+
+/**
+ * This component is used to style the dots (or bullets) used by
+ * react-native-viewpager. It's a purely stateless functional component that is
+ * only concerned with rendering the dots and transitioning the active dot from
+ * one page of the view pager to another.
+ *
+ * @param pageCount {number} The total number of pages.
+ * @param activePage {number} The currently active page.
+ * @param scrollValue {Animated.Value} Used to transition the active dot.
+ * @param scrollOffset {number} Used to reset dot position after a transition.
+ * @return {Component}
+ */
+const ViewPagerIndicator = ({
```

Comment:
This is more than just a shorthand. These stateless functional components are the recommended way to build components that are purely visual. They've been added in React 0.0.14 and they're likely to get an optimisation boost in the next versions. Ideally I'd like all the components in the `components` folder to use this syntax.

You can read more in the [announcement blog](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html) and this [blog post](https://medium.com/@joshblack/stateless-components-in-react-0-14-f9798f8b992d).